p2p/peersHolder/peersHolder.go

```diff
@@ -13,80 +14,106 @@ type peerInfo struct {
 }
 
 type peerIDData struct {
-	connectionAddressSlice string
-	shardID                uint32
-	index                  int
+	connectionAddress string
+	shardID           uint32
+	index             int
 }
 
 type peersHolder struct {
-	preferredConnAddrSlices    []string
-	connAddrSliceToPeerInfo    map[string]*peerInfo
+	preferredConnAddresses     []string
+	connAddrToPeersInfo        map[string][]*peerInfo
 	tempPeerIDsWaitingForShard map[core.PeerID]string
 	peerIDsPerShard            map[uint32][]core.PeerID
 	peerIDs                    map[core.PeerID]*peerIDData
-	sync.RWMutex
+	mut                        sync.RWMutex
 }
 
 // NewPeersHolder returns a new instance of peersHolder
-func NewPeersHolder(preferredConnectionAddressSlices [][]byte) *peersHolder {
+func NewPeersHolder(preferredConnectionAddresses []string) *peersHolder {
 	preferredConnections := make([]string, 0)
-	connAddrSliceToPeerIDs := make(map[string]*peerInfo)
+	connAddrToPeerIDs := make(map[string][]*peerInfo)
 
-	for _, connAddrSlice := range preferredConnectionAddressSlices {
-		preferredConnections = append(preferredConnections, string(connAddrSlice))
-		connAddrSliceToPeerIDs[string(connAddrSlice)] = nil
+	connectionValidator := connectionStringValidator.NewConnectionStringValidator()
+
+	for _, connAddr := range preferredConnectionAddresses {
+		if !connectionValidator.IsValid(connAddr) {
+			continue
```

Comment:
silent error treatment. I think we should return an error in order to signal to the user that his/her input in the prefs.toml file was wrong